### ISSUE TITLE
Battery Percentage no longer showing on many Laptop Types (since 1.0.23)

### ISSUE THREAD

**Issue by NilsLeumann** [Ylianst/MeshCentral#6143]
**Describe the bug**
Since version 1.0.23 and also on version 1.0.24 many laptops that used to show the battery state of charge now no longer show it (no battery Icon, just like the computer doesn't have a battery)
<img width="301" alt="Bildschirmfoto 2024-06-01 um 22 16 47" src="https://github.com/Ylianst/MeshCentral/assets/76911345/59c6cd91-1a5d-4c6c-996f-e97aae1a989f">
. Tested Laptops: Surface Pro 6 (only one affected), ThinkPads (but only a P16, all my E14 AMD show battery just fine), HP ProBooks (only Intel models, so far no AMD)
In my testing it mainly seemed to affect intel based laptops, however I have not yet been able to determine a clear pattern

**To Reproduce**
Well, just log on to meshcentral and check if there is a battery icon :)

**Screenshots**

**Server Software (please complete the following information):**
 - OS: Ubuntu
 - Virtualization: Proxmox
 - Network: WAN
 - Version: 1.0.24

**Client Device (please complete the following information):**
 - Device: Laptop
 - OS: Mac
 - Network: Lan
 - Browser: Chrome

**Remote Device (please complete the following information):**
 - Device: Laptop (Lenovo, HP, Surface, seemingly at random, I have multiple Surfaces (exact same specs) and only one seems to not show batttery percentage
 - OS: Windows 10 (22H2) and Windows 11 (23H2)
 - Network: WAN
 - Current Core Version: December 2022

<img width="301" alt="Bildschirmfoto 2024-06-01 um 22 16 47" src="https://github.com/Ylianst/MeshCentral/assets/76911345/3163897e-fa07-475a-a48e-42a9bf66c0a7">

**Comment by si458** [comment]
yeh ive spotted this recently too and i cant work it out whats happened?
i moved the `power-monitor.js` from meshagent to meshcentral so it fixed the battery tracking for mac laptops https://github.com/Ylianst/MeshCentral/commit/b71b4d04e67175492f608c3b69eb5b2ce2b22804
but it seems to of possibly broke the windows battery?

can you try one thing for me,
delete `agents/modules_meshcore/power-monitor.js` then restart ur meshcentral server?

does the battery levels etc return?

**Comment by si458** [comment]
yes deleting `node_modules/meshcentral/agents/modules_meshcore/power-monitor.js` and restarting meshcentral 
restored the battery levels for laptops, it just loses the battery levels for mac again
so i will do more investigating again

**Comment by NilsLeumann** [comment]
Thank you very much for your quick reply, since I do not manage any Macs in MeshCentral I deleted the power-monitor.js or rather moved it somewhere else
Now it seems to work fine, at least on one of my HP Laptops that didn't show the battery percentage before. I assume all other devices will be fine now but I will see on monday
I also just want to thank you for making and maintaining such an amazing piece of open source software :))

**Comment by si458** [comment]
@NilsLeumann glad it temporarily sorted the issue.
im gunna look into it fully tho as i am confused?
as you said it seemt to show on some but then not others and i dont know how i didnt spot it!
glad u like the software! dont forget to donate ❤️ https://www.si458.co.uk/2024/01/05/donation/

**Comment by si458** [comment]
closing this as i have removed the file from the `master` image and `npm install Ylianst/MeshCentral`
